### ISSUE TITLE
Support the W3C Webdriver spec (?)

### ISSUE THREAD

**Issue by ysbaddaden** [ysbaddaden/selenium-webdriver-crystal#6]
The original [JsonWriteProtocol](https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol) is now obsolete as of Selenium 3.5 and Firefox 55 (at least) and the client should be updated to the [W3C Webdriver spec](https://w3c.github.io/webdriver/webdriver-spec.html) which introduces incompatibilities.

Sadly, the new protocol intermixes everything (driver, proxy, client, ...) and is missing many details about the actual protocol (e.g. what JSON is being transfered as a request and response body?) which makes it hard to follow, especially when compared to the very straightforward [JsonWriteProtocol](https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol) document. Sigh.

**Comment by ysbaddaden** [comment]
The test suite is now passing with Selenium 3.5.3, Geckodriver 0.18.0 and Firefox 55.0.2.

I'm leaving this issue open, since there are probably other changes in the protocol.

**Comment by eliasjpr** [comment]
Just want to let you know that Chromedriver 2.32 and Selenium 3.5.3.

**Comment by ysbaddaden** [comment]
Thanks! I'll keep this open, as there may be some parts of the protocol that changed, but aren't covered by the very minimal test suite.

**Comment by eliasjpr** [comment]
Hey, I would like to talk to you about this project, we the core Amber team are making heavy use it to support system/feature tests. We are considering to make some changes to the driver to fit our needs. Find me on Gitter **eliasjpr** I would like to demo what we have so far.

**Comment by ysbaddaden** [comment]
It seems W3C Webdriver is correctly supported. Let's close.

@eliasjpr feel free to open issues detailing a few issues. I'll look at them.
